servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -7994,6 +8033,117 @@ public DomainList listDomainDependencies(String service) {
         }
     }
 
+    void executeDeleteExpiredMemberships(ResourceContext ctx, ObjectStoreConnection con, int roleId, String roleName,
+                                 int principalId, String normalizedMember, String auditRef, String caller) {
+        try {
+            final String principal = getPrincipalName(ctx);
+
+            // process our delete role member operation
+
+            if (!con.deleteExpiredRoleMember(roleId, principalId, normalizedMember, principal, auditRef)) {
+                throw ZMSUtils.notFoundError(caller + ": unable to delete role member: " +
+                        normalizedMember + " from role: " + roleName, caller);
+            }
+
+            return;
+
+        } catch (ResourceException ex) {
+            throw ex;
+        }
+    }
+
+    public void executeDeleteAllExpiredRoleMemberships(ResourceContext ctx, String auditRef, String caller) {
+        final int maxDbCallsPerRun = new DynamicConfigInteger(CONFIG_MANAGER, ZMSConsts.ZMS_PROP_PURGE_TASK_MAX_DB_CALLS_PER_RUN, ZMSConsts.PURGE_TASK_MAX_DB_CALLS_PER_RUN_DEF).get();
```

Comment:
we should not create dynamic config objects within the method. we can create them in the DBService constructor and then use them here with .get()s.